### PR TITLE
[VarDumper] Added support for amqp

### PR DESCRIPTION
**Pull request by lyrixx**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

screenshot3

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
:+1:

File: src/Symfony/Component/VarDumper/Caster/AmqpCaster.php [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        if (method_exists($c, 'getReadTimeout')) {
+            $timeout = $c->getReadTimeout();
+        } else {
+            $timeout = $c->getTimeout();
+        }
+
+        $a += array(
+            $prefix.'isConnected' => $c->isConnected(),
+            $prefix.'login' => $c->getLogin(),
+            $prefix.'password' => $c->getPassword(),
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Caster/AmqpCaster.php]: should we really put the password ?

Comment (lyrixx) [inline src/Symfony/Component/VarDumper/Caster/AmqpCaster.php]: I dont know. But this feature should be used only in Dev env , so ...

Comment (cordoval) [inline src/Symfony/Component/VarDumper/Caster/AmqpCaster.php]: people will use it in production, I have seen the dumper being actually effective in prototype MVPs. Dumper does not necessarily mean production imo.

**Comment by romainneutron** [comment]
:+1:

File: src/Symfony/Component/VarDumper/Caster/AmqpCaster.php (hunk 1 above)
Comment (lyrixx) [inline src/Symfony/Component/VarDumper/Caster/AmqpCaster.php]: But by default the dumper in only available in `dev` env. so it means people left the `app_dev.php`. and in the WPT you can also find some nice things...

Comment (lyrixx) [inline src/Symfony/Component/VarDumper/Caster/AmqpCaster.php]: And BTW, if you `var_dump($container)` ; nothing prevent you to dump all your passwords in parameter...

Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/AmqpCaster.php]: Yes we should put the password, it's part of the state and we dump the state :)

File: src/Symfony/Component/VarDumper/Caster/AmqpCaster.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        return $a;
+    }
+
+    private static function extractFlags($flags)
+    {
+        $flagsArray = array();
+
+        foreach (self::$flags as $value => $name) {
+            if ($flags & $value) {
```
Comment (aitboudad) [inline src/Symfony/Component/VarDumper/Caster/AmqpCaster.php]: it should be ``` if ($flags && $value) { ``` ?

Comment (aitboudad) [inline src/Symfony/Component/VarDumper/Caster/AmqpCaster.php]: just checked and it's ok, so forgot my last comment :)

**Comment by stof** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @lyrixx.

**Pull request merged**